src/main/kotlin/net/atos/zac/enkelvoudiginformatieobject/EnkelvoudigInformatieObjectLockService.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package net.atos.zac.enkelvoudiginformatieobject
+
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import jakarta.persistence.EntityManager
+import jakarta.persistence.PersistenceContext
+import jakarta.transaction.Transactional
+import jakarta.transaction.Transactional.TxType.REQUIRED
+import jakarta.transaction.Transactional.TxType.SUPPORTS
+import net.atos.client.zgw.drc.DrcClientService
+import net.atos.client.zgw.zrc.ZRCClientService
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.zac.enkelvoudiginformatieobject.model.EnkelvoudigInformatieObjectLock
+import net.atos.zac.util.UriUtil
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+import java.util.Optional
+import java.util.UUID
+
+@ApplicationScoped
+@Transactional(SUPPORTS)
+@AllOpen
+@NoArgConstructor
+class EnkelvoudigInformatieObjectLockService @Inject constructor(
+    private val drcClientService: DrcClientService,
+    private val zrcClientService: ZRCClientService
+) {
+    @PersistenceContext(unitName = "ZaakafhandelcomponentPU")
+    private lateinit var entityManager: EntityManager
```

Comment:
is it not possible to pull this into the constructor?

Reply:
This does not get injected as it is `@PersistenceContext`

@edgarvonk might have more experience here

Reply:
Indeed, it does not get injected. This is the only way it will work as far as I know. And to make it possible to mock it in a unit test, the only solution I could think of is this:

```  /**
     * Sets the entity manager for this service.
     * Only meant for testing purposes! In normal usage the entity manager is injected by the CDI container.
     *
     * @param entityManager the entity manager to set
     */
    fun setEntityManager(entityManager: EntityManager) {
        this.entityManager = entityManager
    }```
    
    See SignaleringService.
    
    I also Googled for it but I think there is no other viable solution.